src/openapi_parser/builders/content.py

```diff
@@ -22,8 +28,8 @@ def build_list(self, data: dict) -> list[Content]:
 
     def _create_content(self, content_type: str, content_value: dict) -> Content:
         logger.debug(f"Content building [type={content_type}]")
-
+        ContentTypeCls: ContentTypeType = ContentType if self.strict_enum else LooseContentType
```

Comment:
`mypy` isn't smart enough yet to figure out that `strict_enum` is a bool so the resulting type is the union of both possible types, so therefore the explicit type hint.

Maybe this could be picked up automatically by creating the instances instead of the classes - I haven't tried it.

Reply:
what was the error?

Reply:
off the top of my head - `object of type object cannot be assigned to...` - this happens because with the `if/else` statement, mypy takes the common ancestor of both options, which in the case of `ContentType` and `LooseContentType` is `object` since they don't subclass the same things.